src/project/qml/MuseScore/Project/internal/PropertyItem.qml

```diff
@@ -0,0 +1,110 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2021 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+import QtQuick 2.15
+import QtQuick.Layouts 1.15
+
+import MuseScore.Ui 1.0
+import MuseScore.UiComponents 1.0
+
+RowLayout {
+    id: root
+
+    width: 100
+    spacing: 8
+
+    property int propertyNameWidth: -1
+    property int propertyValueWidth: 100
+    property NavigationPanel navigationPanel: null
+    property var propertyItem
+
+    signal changePositionOfIndex(int index)
+    signal propertyNameChanged(string newValue)
+    signal propertyValueChanged(string newValue)
+
+    StyledTextLabel {
+        Layout.preferredWidth: root.propertyNameWidth
+
+        text: root.propertyItem.propertyName ? root.propertyItem.propertyName : ""
+        font: ui.theme.bodyBoldFont
+        horizontalAlignment: Qt.AlignLeft
+        visible: root.propertyItem.isStandardProperty ? root.propertyItem.isStandardProperty : false
+    }
+
+    TextInputField {
+        Layout.preferredWidth: root.propertyNameWidth
+
+        currentText: root.propertyItem.propertyName ? root.propertyItem.propertyName : ""
+        visible: !root.propertyItem.isStandardProperty
+        hint: qsTrc("project", "Property")
+
+        navigation.name: root.propertyItem.propertyName + "PropertyName"
+        navigation.panel: root.navigationPanel
+        navigation.column: root.propertyItem.index * 3
```

Comment:
I don't quite understand what this root.propertyItem.index * 3 means. Could you please explain it?

Reply:
OK, Let's add a private property:

```
QObject {
    id: prv
    
    property int navigationStartIndex: root.propertyItem.index * 3
}
```

And will reuse it for each part of the component:

```
TextInputField {
    navigation.column: prv.navigationStartIndex
    ...
}

TextInputField {
    navigation.column: prv.navigationStartIndex + 1
    ...
}
...
```